gitpod-io/gitpod
# Fix JetBrains warmup configured `both` versions but only latest version takes effect

## Description


## Related Issue(s)

Relates ENT-253

## How to test

Join my org https://hw-jb-warmup.preview.gitpod-dev.com/orgs/join?inviteId=5ab0c601-7e4a-4e45-af2d-0cc0813abf68
- Start workspace for https://github.com/mustard-mh/spring-petclinic
- check if there're stable and latest versions of caches
```
ls /workspace/.cache/
```

✅ <img width="936" alt="image" src="https://github.com/gitpod-io/gitpod/assets/20944364/066415b8-440f-48e1-ae34-4ba1502f0c3a">

**Or create org and prebuild yourself**
- Add repository https://github.com/gitpod-samples/spring-petclinic (fork if you can't enable prebuild for it)
- Trigger a prebuild on main branch
- Start workspace from it and check if there're stable and latest versions of caches
```
ls /workspace/.cache/
```

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-jb-warmup</li>
	<li><b>🔗 URL</b> - <a href="https://hw-jb-warmup.preview.gitpod-dev.com/workspaces" target="_blank">hw-jb-warmup.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-jb-warmup-gha.25948</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-jb-warmup%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
